### PR TITLE
add missing Xcode requirements to GSG

### PR DESCRIPTION
**Pull request by Azulinho**
doc fix for:
https://clusterhq.atlassian.net/browse/FLOC-2253

### CODE REVIEW

**Comment by adamtheturtle** [comment]
This is also a problem at http://doc-dev.clusterhq.com/using/installing/index.html#os-x.

Do we need the whole of Xcode? Can we just use the command line tools? (`xcode-select --install`). If the latter then can we put that command into our testing infrastructure (can be a follow-up)? Or does it need a user to click a button on pop-up.

Usually I would say that we should have a fresh OS X machine and test these instructions without Xcode, then a fresh one with Xcode and see, but it doesn't seem useful now. Please resubmit after addressing.

**Comment by Azulinho** [comment]
just updated using/installing/ to include the Xcode installation reference.

Can't anwser if whole of Xcode is needed or just a subset, nor if the command line tools can be used.
TBH not being a OSX user I am not entirelly sure what's contained in the Xcode package. 
In my opinion it is not worth to test out what bit of Xcode is needed (at least not in relation to this JIRA).

Regarding adding the Xcode step to CI, it should be discussed in a separate issue.

**Comment by adamtheturtle** [comment]
Installing Xcode includes downloading multiple GBs and needs the user to select things in a different app, taking them away from the guide more than they already are. I think that it is worth the time to test if possible.

Have you tested this fix?

> Regarding adding the Xcode step to CI, it should be discussed in a separate issue.

Agreed. Please create this issue.

File: docs/using/installing/index.rst [hunk 1]
```diff
@@ -110,6 +110,8 @@ You'll need to do this every time you start a new shell.
 OS X
 ----
 
+Install `Xcode` from the Mac App Store.
```
Comment (adamtheturtle) [inline docs/using/installing/index.rst]: Either use double backticks, or preferably use a link:

```
`Xcode`_

(then later in the document)

.. _`Xcode`: https://example.com/xcode
```

**Comment by Azulinho** [comment]
"Installing Xcode includes downloading multiple GBs and needs the user to select things in a different app, taking them away from the guide more than they already are. I think that it is worth the time to test if possible."

@adamtheturtle  this comment is more aligned to discussing if the requirement for installing Xcode should be in the guides than actually the steps for installing it.

**Comment by adamtheturtle** [comment]
No sorry I didn't explain myself - I believe that command installs only the subset of Xcode needed for this - a much much smaller download.

**Comment by adamtheturtle** [comment]
That is untested

**Comment by lukemarsden** [comment]
Doesn't brew depend on Xcode already?

**Comment by Azulinho** [comment]
@lukemarsden looks like it doesn't depend on it anymore

**Comment by Azulinho** [comment]
Sorry, not able to test this:

```vagrant-osx:~ vagrant$ xcode-select --install
xcode-select: error: no developer tools were found, and no install could be requested (perhaps no UI is present), please install manually from 'developer.apple.com'.
```

I am not able to download the .dmg from the website from my laptop, I always get re-directed to a page to upgrade to yosemite.

**Comment by davidoliverSP2** [comment]
This change affects the new GSG (FLOC 1920): 
PR: https://github.com/ClusterHQ/flocker/pull/1506
Doc: http://build.clusterhq.com/results/docs/newdoc-tryflocker-FLOC-1920/build-7963/using/tutorial/moving-volume-tutorial.html

The doc changed on this PR will 100% *not* be in v1.0 according to Marketing. (The option is either 1920 +/- TrueAbility hosted environment tutorial, or 1920 alone - there is no scenario where we want to publish the pre1.0 Getting Started Guide).

**Comment by Azulinho** [comment]
thankz @davidoliverSP2 this is a throwaway then

**Comment by adamtheturtle** [comment]
Sure but the change is still relevant to the new guide.

**Comment by ndtreviv** [comment]
I've run the command in an attempt to install flocker on my mac - it only installs a subset of X-Code and fixes the problem.
However, there is now another problem related to a recent release of openssl:

`building '_openssl' extension
creating build/temp.macosx-10.11-intel-2.7/build
creating build/temp.macosx-10.11-intel-2.7/build/temp.macosx-10.11-intel-2.7
clang -DNDEBUG -g -fwrapv -Os -Wall -Wstrict-prototypes -I/usr/local/opt/flocker-1.10.2/openssl/include -arch x86_64 -arch i386 -pipe -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c build/temp.macosx-10.11-intel-2.7/_openssl.c -o build/temp.macosx-10.11-intel-2.7/build/temp.macosx-10.11-intel-2.7/_openssl.o
build/temp.macosx-10.11-intel-2.7/_openssl.c:686:6: error: conflicting types for 'BIO_new_mem_buf'
BIO *BIO_new_mem_buf(void *, int);
     ^
/usr/local/opt/openssl/include/openssl/bio.h:692:6: note: previous declaration is here
BIO *BIO_new_mem_buf(const void *buf, int len);
     ^
1 error generated.
error: command 'clang' failed with exit status 1` 

Maybe related to this: https://github.com/pyca/cryptography/issues/2750

I notice that the brew recipe installs cryptography:
`==> Downloading https://pypi.python.org/packages/source/c/cryptography/cryptography-1.1.2.tar.gz`
Does this need to be updated?

Should I raise this as another issue, or am I totally off the mark here?

**Comment by ndtreviv** [comment]
Because I want this to work, I took the liberty of forking the brew tap and updating it.
Think I've fixed it all here: https://github.com/ndtreviv/homebrew-tap/blob/master/flocker-1.10.2.rb

The brew install runs. Question is, did the install work!?

**Comment by gianpaj** [comment]
I'm having the same issue.
```
$ clang --version
Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin14.5.0
Thread model: posix


$ brew install flocker-1.10.2
...
==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/flocker-1.10.2
Last 15 lines from /Users/gianfranco/Library/Logs/Homebrew/flocker-1.10.2/14.python:
generating cffi module 'build/temp.macosx-10.10-intel-2.7/_padding.c'
generating cffi module 'build/temp.macosx-10.10-intel-2.7/_constant_time.c'
generating cffi module 'build/temp.macosx-10.10-intel-2.7/_openssl.c'
building '_openssl' extension
creating build/temp.macosx-10.10-intel-2.7/build
creating build/temp.macosx-10.10-intel-2.7/build/temp.macosx-10.10-intel-2.7
clang -DNDEBUG -g -fwrapv -Os -Wall -Wstrict-prototypes -I/usr/local/opt/flocker-1.10.2/openssl/include -arch x86_64 -arch i386 -pipe -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c build/temp.macosx-10.10-intel-2.7/_openssl.c -o build/temp.macosx-10.10-intel-2.7/build/temp.macosx-10.10-intel-2.7/_openssl.o
build/temp.macosx-10.10-intel-2.7/_openssl.c:686:6: error: conflicting types for 'BIO_new_mem_buf'
BIO *BIO_new_mem_buf(void *, int);
     ^
/usr/local/opt/openssl/include/openssl/bio.h:692:6: note: previous declaration is here
BIO *BIO_new_mem_buf(const void *buf, int len);
     ^
1 error generated.
error: command 'clang' failed with exit status 1
...
```

I didn't manage to find how to fix it by hacking it the @ndtreviv 's way

**Comment by wallnerryan** [comment]
Thanks @gianpaj and @ndtreviv ! I can verify the same issue, I'm having the team take a look at this now. Thanks for reporting!

**Comment by ndtreviv** [comment]
Great! As I said: I just had to upgrade cryptography (to 1.2.3) and also cffi (to 1.5.2) and change requirements.txt in the deployment accordingly. 

If this doesn't work for others, I'd suggest they clear their brew download cache and try again.

**Comment by ndtreviv** [comment]
Or I can post my full brew recipe with my modified flocker download if that helps...

**Comment by ndtreviv** [comment]
I've posted my full recipe:
https://github.com/ndtreviv/homebrew-tap/blob/master/flocker-1.10.2.rb

So, if people want a working one RIGHT NOW they can:
```
brew tap ndtreviv/tap
brew install ndtreviv/tap/flocker-1.10.2
```
Note: homebrew will download resources into a Cache directory. Make sure that if you've run this or the ClusterHQ one before that you've cleared those caches!

**Comment by gianpaj** [comment]
that worked! thanks @ndtreviv - faster than ClusterHQ haha

The brew test command works.

FYI: not sure if related, but I'm getting an issue when running `flocker-diagnostics`:
```
$ flocker-diagnostics
Traceback (most recent call last):
  File "/usr/local/Cellar/flocker-1.10.2/1.10.2/libexec/bin/flocker-diagnostics", line 9, in <module>
    load_entry_point('Flocker==1.10.2', 'console_scripts', 'flocker-diagnostics')()
  File "/usr/local/Cellar/flocker-1.10.2/1.10.2/libexec/vendor/lib/python2.7/site-packages/pkg_resources/__init__.py", line 558, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/local/Cellar/flocker-1.10.2/1.10.2/libexec/vendor/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2682, in load_entry_point
    return ep.load()
  File "/usr/local/Cellar/flocker-1.10.2/1.10.2/libexec/vendor/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2355, in load
    return self.resolve()
  File "/usr/local/Cellar/flocker-1.10.2/1.10.2/libexec/vendor/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2361, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/Cellar/flocker-1.10.2/1.10.2/libexec/lib/python2.7/site-packages/flocker/node/__init__.py", line 11, in <module>
    from ._deploy import (
  File "/usr/local/Cellar/flocker-1.10.2/1.10.2/libexec/lib/python2.7/site-packages/flocker/node/_deploy.py", line 14, in <module>
    from ..control._model import NodeState
  File "/usr/local/Cellar/flocker-1.10.2/1.10.2/libexec/lib/python2.7/site-packages/flocker/control/__init__.py", line 25, in <module>
    from ._protocol import (
  File "/usr/local/Cellar/flocker-1.10.2/1.10.2/libexec/lib/python2.7/site-packages/flocker/control/_protocol.py", line 48, in <module>
    from repoze.lru import LRUCache
ImportError: No module named repoze.lru
```

**Comment by wallnerryan** [comment]
hey @gianpaj and @ndtreviv, we're working on a fix natively, sorry for the delay its just lower on our priority list :).

The `flocker-diagnostics` tool is mean to be run on a flocker node, not osx. So yes, unrelated.

**Comment by sarum90** [comment]
Thanks @ndtreviv for the fix!

Incorporated into #2703, should be in our official homebrew tap soon.

**Comment by jtyberg** [comment]
I hit this too on OS X 10.11.3.

I followed the @ndtreviv path and used a modified recipe to get the brew install to work.

FWIW, I had trouble finding the `requires.txt` file after install to update the versions of `cffi` and `cryptography`.  Mine was here:

```
/usr/local/Cellar/flocker-1.10.2/1.10.2/libexec/lib/python2.7/site-packages/Flocker-1.10.2-py2.7.egg-info/requires.txt
```

**Pull request closed without merging**